packages/grid/_modules_/grid/hooks/features/rows/useGridParamsApi.ts

```diff
@@ -100,12 +99,15 @@ export function useGridParamsApi(apiRef: GridApiRef) {
   const getCellValue = React.useCallback(
     (id: GridRowId, field: string) => {
       const colDef = apiRef.current.getColumnFromField(field);
-      const rowModel = apiRef.current.getRowFromId(id);
 
-      if (!colDef) {
-        warnMissingColumn(field);
+      if (!warnedOnce && process.env.NODE_ENV !== 'production') {
+        if (!colDef) {
```

Comment:
why the nested if?

Reply:
The outer one is for the prod vs. dev mode. The inner one is for the actual logic.

Reply:
I would start with the production check and put it in 1 if. 

but if you take your route then warnedOnce should go to the second if

Reply:
warnedOnce is first so that it resolves faster once it has triggered once in dev, it has no impact in production, the whole branch is removed.

Reply:
Yes I know hence why I would put the production first...
But not important...

Reply:
Consider the following, in dev mode, with the current order

```jsx
 if (false && true) { // has never warned
 if (true && true) { // has already warned
 ```
 
and with the opposite order

```jsx
 if (true && false) { // has never warned
 if (true && true) { // has already warned
 ```
 
assuming true and false are actually computations. What would you rather have? I would personally pick the first version as it has an early exit.